Libs/Professions/professions.lua

```diff
@@ -0,0 +1,137 @@
+            type = "header",
+            name = "Professions",
+        },
+        description = {
+            order = 1,
+            type = "description",
+            name = function() return Professions:Description() end,
+        },
+        selectprofession = {
+            order = 2,
+            type = "select",
+            name = "Select Profession",
+            values = {
+                Alchemy = "Alchemy",
+                Blacksmithing = "Blacksmithing",
+                Enchanting = "Enchanting",
+                Leatherworking = "Leatherworking",
+                Tailoring = "Tailoring"
+            },
+            get = function(info)
+                return selectedProfession;
+            end,
+            set = function(info, value)
+                Professions:updateOptions(value);
+            end,
+        },
+        selectsubprof = {
+            order = 3,
+            type = "select",
+            name = "Category",
+            values = subProfessionValues;
+            get = function(info)
+                return selectedSubProfession;
+            end,
+            set = function(info, value)
+                Professions:updateSubOptions(value);
+            end,
+        },
+        professors = {
+            order = 4,
+            type = "description",
+            name = function() return Professions:updateProfessors() end;
+        }
+    }
+}
+
+function Professions:updateOptions(value)
+    selectedProfession = value;
+    selectedSubProfession = "";
+    count = #subProfessionValues
+    for i=0, count do subProfessionValues[i]=nil end
+    if value == "Alchemy" then
+        table.insert(subProfessionValues, "Flasks");
+        table.insert(subProfessionValues, "Elixirs");
+        table.insert(subProfessionValues, "Resist Potions");
+        table.insert(subProfessionValues, "Miscellaneous");
+    elseif value == "Blacksmithing" then
+        table.insert(subProfessionValues, "Armor");
+        table.insert(subProfessionValues, "Weapon");
+        table.insert(subProfessionValues, "Miscellaneous");
+    elseif value == "Enchanting" then
+        table.insert(subProfessionValues, "Boots");
+        table.insert(subProfessionValues, "Bracer");
+        table.insert(subProfessionValues, "Chest");
+        table.insert(subProfessionValues, "Cloak");
+        table.insert(subProfessionValues, "Gloves");
+        table.insert(subProfessionValues, "Shield");
+        table.insert(subProfessionValues, "Weapon");
+        table.insert(subProfessionValues, "Oils");
+    elseif value == "Leatherworking" then
+        table.insert(subProfessionValues, "Armor");
+        table.insert(subProfessionValues, "Cloak");
+    elseif value == "Tailoring" then
+        table.insert(subProfessionValues, "Bags");
+        table.insert(subProfessionValues, "Armor");
+        table.insert(subProfessionValues, "Cloak");
+    end
+end
+
+function Professions:updateSubOptions(value)
+    selectedSubProfession = value;
```

Comment:
I wonder if we can remove this function and just add ` selectedSubProfession = value `  to line 52

Reply:
oops yes we can